arch/risc-v/src/common/riscv_percpu.h

```diff
@@ -39,9 +39,9 @@
  ****************************************************************************/
 
 #ifdef __ASSEMBLY__
-#define SCRATCH_HARTID_OFFSET   (0 * INT_REG_SIZE)
+#define RISCV_HARTID_OFFSET   (0 * INT_REG_SIZE)
 #else
-#define SCRATCH_HARTID_OFFSET   offsetof(riscv_percpu_s, hartid)
+#define RISCV_HARTID_OFFSET   offsetof(riscv_percpu_s, hartid)
```

Comment:
What about??
```
#define RISCV_HARTID_OFFSET (0)
#ifndef __ASSEMBLY__
_Static_assert(RISCV_HARTID_OFFSET == offsetof(riscv_percpu_s, hartid), "Error");
#ensif
```
I'm not sure if it works. Maybe static assert can be placed in C file.

Reply:
IMO the prefix should pre PERCPU_ like the file name. The original name before I published this was riscv_scratch.c/h but I changed it and forgot to fix the names.

Reply:
Both done.